packages/myst-transforms/src/html.ts

```diff
@@ -69,6 +73,85 @@ export function htmlTransform(tree: GenericParent, opts?: HtmlTransformOptions)
   return tree;
 }
 
+/**
+ * Convert html nodes and children to single html node
+ *
+ * This function takes the html nodes with opening and closing tags; all the mdast content
+ * between these nodes is present as 'children' on the opening node. The mdast content is
+ * then converted to html, wrapped by the opening and closing tag, and saved to a single html
+ * node. All of the processed nodes are then marked for deletion.
+ */
+function finalizeNode(htmlOpenNodeWithChildren: GenericParent, htmlCloseNode: GenericNode) {
+  const innerHtml = mystToHtml(
+    { type: 'root', children: htmlOpenNodeWithChildren.children },
+    {
+      hast: {
+        handlers: {
+          html: (h, node) => {
+            return fromHtml(node.value, { fragment: true }).children as ElementContent[];
+          },
+        },
+      },
+    },
+  );
+  // This would be good to sanitize, but the best solution requires jsdom, increasing build size by 50%...
```

Comment:
@stevejpurves `dompurify` requires `jsdom` which bumps our `mystmd` bundle size from ~12mb -> ~17mb. Feels not great. We will need it on the other side, where this html is consumed (`jupyterlab-myst` or `myst-to-react`).

Another option is `sanitize-html`. This is a simpler library, but we need to explicitly decide which html tags/attrs to allow, and I don't think we have enough info for that yet. (For example, the defaults do not allow `button`, which is needed for this use case: https://github.com/executablebooks/jupyterlab-myst/issues/64https://github.com/executablebooks/jupyterlab-myst/issues/64)